alexandria/core/views.py

```diff
@@ -136,31 +137,25 @@ def convert(self, request, pk=None):
             request.user, settings.ALEXANDRIA_CREATED_BY_GROUP_PROPERTY, None
         )
 
-        converted_document = models.Document.objects.create(
-            title={k: f"{ splitext(v)[0]}.pdf" for k, v in document.title.items()},
-            description=document.description,
-            category=document.category,
-            date=document.date,
-            metainfo=document.metainfo,
-            created_by_user=username,
-            created_by_group=group,
-            modified_by_user=username,
-            modified_by_group=group,
-        )
         file_name = f"{splitext(file.name)[0]}.pdf"
-        converted_file = models.File.objects.create(
-            document=converted_document,
-            name=file_name,
-            content=ContentFile(response.content, file_name),
-            mime_type="application/pdf",
-            size=len(response.content),
-            metainfo=file.metainfo,
-            created_by_user=username,
-            created_by_group=group,
-            modified_by_user=username,
-            modified_by_group=group,
+        converted_document, __ = create_document_file(
+            username,
+            group,
+            document.category,
+            {k: f"{ splitext(v)[0]}.pdf" for k, v in document.title.items()},
+            file_name,
+            ContentFile(response.content, file_name),
+            "application/pdf",
+            len(response.content),
+            {
+                "description": document.description,
+                "date": document.date,
+                "metainfo": document.metainfo,
+            },
+            {
+                "metainfo": file.metainfo,
+            },
```

Comment:
please use named arguments when there are more than 2-3 parameters. It becomes too easy to make mistakes otherwise

Reply:
Done